src/components/HistoryTables/MessageHistory/Detail/index.tsx

```diff
@@ -73,25 +74,28 @@ export default function MessageDetail(props: MessageDetailProps) {
 
   const execReturn = useMemo<ExecReturn | null>(() => {
     if (!message) return null
-    const toExec = isAddrEqual(message?.to, 'f01')
-    const receiptExists = !!msgRcptQuery?.receipt?.return
-    const initTx = Number(message?.method) === 2
-    if (toExec && receiptExists && initTx) {
-      return getAddrFromReceipt(msgRcptQuery?.receipt?.return)
-    }
-    return null
+    const isToExec = isAddrEqual(message?.to, 'f01')
+    const isInitTx = Number(message?.method) === 2
+    const receiptReturn = msgRcptQuery?.receipt?.return
+    return isToExec && isInitTx && receiptReturn
+      ? getAddrFromReceipt(receiptReturn)
+      : null
   }, [message, msgRcptQuery?.receipt])
 
   const messageState = useMemo<MessageState>(() => {
+    if (error) return MessageState.Error
+    if (loading) return MessageState.Loading
     if (pending) return MessageState.Pending
-    else if (loading) return MessageState.Loading
-    else if (!message && !pending && !error) return MessageState.NotFound
-    else if (!message && !pending && !!error) return MessageState.Error
-    else if (!!message && !gasUsed) return MessageState.Confirmed
-    else if (!!message && !!gasUsed) return MessageState.Executed
-    else return MessageState.Loading
+    if (!message) return MessageState.NotFound
+    if (!gasUsed) return MessageState.Confirmed
+    return MessageState.Executed
```

Comment:
I'd simplify this a bit:
- because we `return` after `if`, there is no need to use `else`
- if, for example, we return on `if (pending)`, we never need to check for `if (!pending)` after, pending will always be falsy
- Keeping the above in mind, we can change the order so that one check per line ccovers all the cases. On the final return:
  - we have no error
  - we're not loading
  - we're not pending
  - we have a message
  - we have gas parameters
  meaning the message was successfully executed

Reply:
I prefer a more explicit approach to determining if a message was executed rather than this more "implied" approach.

Reason being is messy and not necessarily "good", but I have a feeling there could be a state where:
```
we have no error
we're not loading
we're not pending
we have a message
we have gas parameters
```

Yet still, for whatever reason, the message's gas costs and receipt have not been populated for one reason or another

Reply:
That I understand, but I wonder what the difference is with the previous approach, as it was also checking the same conditions, but in a different syntax?

I was also wondering if we should check te receipt for determining if the message was actually executed